provisioning/transport/amqp/src/AmqpAuthStrategySymmetricKey.cs

```diff
@@ -41,14 +42,13 @@ public override AmqpSettings CreateAmqpSettings(string idScope)
             return settings;
         }
 
-        public override Task OpenConnectionAsync(AmqpClientConnection connection, TimeSpan timeout, bool useWebSocket, IWebProxy proxy)
+        public override Task OpenConnectionAsync(AmqpClientConnection connection, TimeSpan timeout, bool useWebSocket, IWebProxy proxy, RemoteCertificateValidationCallback remoteCertificateValidationCallback)
         {
```

Comment:
Is this for trusted certs here? why would you need this in symmetric key code path ?

Reply:
The remote certificate validation callback is for validating the remote certificate, which is the certificate presented by the iot hub, in our case. This is the certificate that the server would preset for TLS negotiation. It does not depend on the specific authentication mechanism used for validating the device identity.